src/lib/View/Content/ContentType/ContentTypeParameterSupplier.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\View\Content\ContentType;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\Core\MVC\Symfony\View\ContentView;
+use EzSystems\HybridPlatformUi\View\Content\ParameterSupplier;
+
+/**
+ * Adds parameters needed to display content types.
+ */
+class ContentTypeParameterSupplier implements ParameterSupplier
+{
+    /**
+     * @var ContentTypeService
+     */
+    protected $contentTypeService;
+
+    public function __construct(ContentTypeService $contentTypeService)
+    {
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supply(ContentView $contentView)
+    {
+        $contentType = $this->contentTypeService->loadContentType(
```

Comment:
well not really I think. It seems like it can only throw a NotFoundException and if it's the case that's a very bad sign for the database consistency ;-) and we can not do anything about that so I guess it's ok like this